### DISCUSSION TITLE
Heads up: Version 1.0.0 is slated to be CRSF for Arduino's first official release.

### DISCUSSION THREAD

**Discussion by ZZ-Cat** [ZZ-Cat/CRSFforArduino#48] [Announcements]
Yup. You read that correctly. The upcoming Version 1.0.0 of CRSF for Arduino will be the first ever official release.

Right now, when you click on my [Releases](https://github.com/ZZ-Cat/CRSFforArduino/releases) section, it currently shows you all of my pre-releases.  
Making the upcoming Version 1.0.0 a full release means that the _latest_ release will show up, and when you click on my Releases tab, it will automatically link you to that instead.

#47 has been merged, and CRSF for Arduino now has full telemetry support.  
Development and testing of Version 1.0.0 is being tracked in #61, where I have a to-do list of things before the release can be made.

If you would like to help me out with testing this, you can checkout the development branch [here](https://github.com/ZZ-Cat/CRSFforArduino/tree/Version-100-Development), and report any bugs you find [here](https://github.com/ZZ-Cat/CRSFforArduino/issues).

Thank you all _so_ much from the bottom of my heart, for using CRSF for Arduino.  
It has been a tough year for me, and I appreciate you all for using my library in your projects.

**Comment by AdrienRostaing** [comment]
Hi,

First of all, I would like to congratulate you for your great work. I am working on a robotics project and wanted to create my own remote control containing a crsf transmitter. How difficult would it be for me to adapt your program to my specific use ? I am not exactly a novice in programming but c++ isn't my forte. If you have any pointers, I would be extremly greatful. 

Adrien Rostaing

**Reply by ZZ-Cat** [reply]
Well, for a start, CRSF for Arduino _is_ a firmware _library._  
You bring this into _your_ main source code (IE Your Arduino sketch) and tailor it to suit your project's needs.

It shouldn't be _too_ difficult to do, as I have designed the API to be fairly simple and easy-to-use.

A basic setup looks something along the lines of this:

- **Controller:** RadioMaster TX16S running EdgeTX firmware.
  - **Transmitter Module:** RadioMaster Ranger with the latest ExpressLRS firmware.
- **Receiver:** RadioMaster RP3 Diversity receiver, also running the latest ExpressLRS firmware.

...and the receiver is connected to the default Serial port on a development board (EG Adafruit Metro M4 Express) that's been flashed with firmware that included CRSF for Arduino.
That is what I use to test and develop CRSF for Arduino.

In its current state (IE Being `MAJOR` version 0.x.x), it's considered to be "early access" and in "beta". That means a lot of important features are yet to be implemented (such as full telemetry feedback, MSP support etc), and there may be bugs that have the potential to impede functionality. You _may_ use this in your project in its current state. But, do so **at your own risk.**

If you _are_ using this in your project, and you spot a bug, do not hesitate to open an Issue.
